src/sentry/debug_files/bundle_index.py

```diff
@@ -0,0 +1,77 @@
+from datetime import datetime
+from typing import Dict, List
+
+from sentry.models.artifactbundle import ArtifactBundleArchive, SourceFileType
+
+
+class BundleIndex:
+    def __init__(self) -> None:
+        self.bundles: Dict[int, datetime] = {}
+        self.files_by_url: Dict[str, List[dict]] = {}
+        self.files_by_debug_id: Dict[str, List[dict]] = {}
```

Comment:
I think that there are two ways of representing this (under the assumption that we don't mix debug ids and urls in the same index):
* Have the system automatically infer debug id or url based on the indexing key.
* Have it like you suggested with self describing keys in the json.

Reply:
What do you think?